output/compact.go

```diff
@@ -137,7 +138,29 @@ func submitCompactSnapshot(ctx context.Context, server *state.Server, collection
 	if len(msg) > 0 && collectionOpts.TestRun {
 		logger.PrintInfo("  %s", msg)
 	} else if !quiet {
-		logger.PrintInfo("Submitted compact %s snapshot successfully", kind)
+		logger.PrintVerbose("Submitted compact %s snapshot successfully", kind)
+		if server.CompactLogTime.IsZero() {
+			server.CompactLogTime = time.Now()
+			server.CompactLogStats = make(map[string]uint8)
+		} else if time.Now().Sub(server.CompactLogTime) > time.Minute {
+			server.CompactLogTime = time.Now()
```

Comment:
I intentionally didn't do that, to avoid clock skew from the time it takes to build the log message. But it should never take a long time and this code path is non-critical so I'll move it down to improve readability.

Reply:
🤔 a way to avoid the timestamp changing over time would be to truncate it to the nearest even minute, instead of simply using `time.Now()`

Reply:
This would ensure we always print the log at an even minute. Though it does mean that when initially starting up the collector you may have to wait was long as 1m59s before anything is logged.

```diff
@@ -140,7 +140,7 @@ func submitCompactSnapshot(ctx context.Context, server *state.Server, collection
        } else if !quiet {
                logger.PrintVerbose("Submitted compact %s snapshot successfully", kind)
                if server.CompactLogTime.IsZero() {
-                       server.CompactLogTime = time.Now()
+                       server.CompactLogTime = time.Now().Truncate(time.Minute)
                        server.CompactLogStats = make(map[string]uint8)
                } else if time.Now().Sub(server.CompactLogTime) > time.Minute {
                        var keys []string
@@ -158,7 +158,7 @@ func submitCompactSnapshot(ctx context.Context, server *state.Server, collection
                        if len(details) > 0 {
                                logger.PrintInfo("Compact snapshots submitted: " + details)
                        }
-                       server.CompactLogTime = time.Now()
+                       server.CompactLogTime = time.Now().Truncate(time.Minute)
                        server.CompactLogStats = make(map[string]uint8)

Reply:
Example output:
```
./pganalyze-collector
2023/11/08 15:50:00 I [server1] Compact snapshots submitted: 5 activity
2023/11/08 15:50:00 I [server1] Submitted snapshot successfully
2023/11/08 15:51:00 I [server1] Compact snapshots submitted: 5 activity
2023/11/08 15:52:00 I [server1] Compact snapshots submitted: 5 activity
2023/11/08 15:53:00 I [server1] Compact snapshots submitted: 5 activity, 1 logs
2023/11/08 15:54:00 I [server1] Compact snapshots submitted: 5 activity, 1 logs
2023/11/08 15:55:00 I [server1] Compact snapshots submitted: 5 activity

Reply:
I went ahead and added this change; it's nice that the logs show up reliably at the same time every minute.